### PR TITLE
ACS-4033: List linked categories for a particular node

### PR DESCRIPTION
**Pull request by krdabrowski**
https://alfresco.atlassian.net/browse/ACS-4033
GET /nodes/{nodeId}/category-links

### CODE REVIEW

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/CategoriesRestTest.java [hunk 1]
```diff
@@ -88,4 +118,18 @@ protected RestCategoryModel createCategoryModelWithIdAndName(final String id, fi
             .name(name)
             .create();
     }
+
+    protected RestCategoryLinkBodyModel createCategoryLinkModelWithId(final String id)
+    {
+        final RestCategoryLinkBodyModel categoryLinkModel = new RestCategoryLinkBodyModel();
+        categoryLinkModel.setCategoryId(id);
+        return categoryLinkModel;
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/CategoriesRestTest.java:126]: I thought we had builders for TAS model objects now :)

File: remote-api/src/main/java/org/alfresco/rest/api/Categories.java [hunk 2]
```diff
@@ -53,6 +53,13 @@
 
     void deleteCategoryById(String id, Parameters parameters);
 
+    /**
+     *
+     * @param nodeId
+     * @return
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/Categories.java:59]: Very minimal Javadoc! Probably better to delete this if you're not going to explain what it does.

**Review by tpage-alfresco — APPROVED** [review]
LGTM

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/CategoriesRestTest.java (hunk 1 above)
Comment (krdabrowski) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/CategoriesRestTest.java:126]: We should :) I wasn't sure if it pays off to create a builder for one field... but you are right - we shouldn't be lazy - I will add the builder 😅

File: remote-api/src/main/java/org/alfresco/rest/api/Categories.java (hunk 2 above)
Comment (krdabrowski) [reply remote-api/src/main/java/org/alfresco/rest/api/Categories.java:59]: Interfaces... so easy to miss them - I've forgot to fill that. Will be fixed in a second.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java [hunk 3]
```diff
@@ -195,11 +214,19 @@ private void verifyAdminAuthority()
...
+            throw new PermissionDeniedException(NO_PERMISSION_TO_READ_CONTENT);
+        }
+    }
+
     private void verifyChangePermission(final NodeRef nodeRef)
     {
         if (permissionService.hasPermission(nodeRef, CHANGE_PERMISSIONS) != ALLOWED)
         {
-            throw new PermissionDeniedException(NO_PERMISSION_TO_READ_CONTENT);
+            throw new PermissionDeniedException(NO_PERMISSION_TO_CHANGE_CONTENT);
```
Comment (mpichura) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java:229]: Good catch

**Review by mpichura — APPROVED** [review]
LGTM. 
I've had some questions regarding method naming but they're minor.

File: remote-api/src/main/java/org/alfresco/rest/api/Categories.java [hunk 4]
```diff
@@ -53,6 +53,13 @@
...
+     */
+    List<Category> getLinkedToCategories(String nodeId);
```
[the remark this answers is not in the document]
Comment (krdabrowski) [reply remote-api/src/main/java/org/alfresco/rest/api/Categories.java:63]: I will adjust the name.

File: packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/requests/Node.java [hunk 5]
```diff
@@ -1112,6 +1112,17 @@ public RestRuleExecutionModel executeRules(RestRuleExecutionModel body)
         return restWrapper.processModel(RestRuleExecutionModel.class, request);
     }
 
+    /**
+     * Get linked categories performing GET cal on "/nodes/{nodeId}/category-links"
+     *
+     * @return categories which are linked from content
+     */
+    public RestCategoryModelsCollection getLinkedToCategories()
```
[the remark this answers is not in the document]
Comment (krdabrowski) [reply packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/requests/Node.java:1120]: We link content to categories, so maybe we should simplify it: `usingNode(file).getLinkedCategories()`?

**Pull request merged**